packages/scoutgame/src/users/mergeUserAccount.ts

```
@@ -0,0 +1,215 @@
+
+export type ProfileToKeep = 'current' | 'new';
+
+export const mergeUserAccount = async ({
+  userId,
+  farcasterId,
+  telegramId,
+  selectedProfile
+}: {
+  userId: string;
+  farcasterId?: number;
+  telegramId?: number;
+  selectedProfile: ProfileToKeep;
+}) => {
+  if (!farcasterId && !telegramId) {
+    throw new Error('No account identities to merge');
+  }
+
+  const [primaryUser, secondaryUser] = await Promise.all([
+    prisma.scout.findUniqueOrThrow({
+      where: {
+        id: userId
+      },
+      select: {
+        builderStatus: true
+      }
+    }),
+    prisma.scout.findFirstOrThrow({
+      where: farcasterId ? { farcasterId } : { telegramId },
+      select: {
+        builderStatus: true,
+        id: true
+      }
+    })
+  ]);
+
+  if (primaryUser.builderStatus !== null && secondaryUser.builderStatus !== null) {
+    throw new Error('Can not merge two builder accounts');
+  }
+
+  // The id of the user to retain
+  const retainedUserId = secondaryUser.builderStatus !== null ? secondaryUser.id : userId;
+
+  // The id of the user to merge into the retained user
+  // The merged account must not be a builder
+  const mergedUserId = secondaryUser.builderStatus === null ? secondaryUser.id : userId;
+
+  const mergedUser = await prisma.scout.findUniqueOrThrow({
+    where: {
+      id: mergedUserId
+    },
+    select: {
+      id: true,
+      farcasterName: true,
+      builderStatus: true,
+      walletENS: true,
+      avatar: true,
+      bio: true,
+      displayName: true,
+      email: true
+    }
+  });
+
+  await prisma.$transaction(
+    async (tx) => {
+      const updatedUserData: Prisma.ScoutUpdateInput = {
+        farcasterId,
+        farcasterName: farcasterId ? mergedUser.farcasterName : undefined,
+        telegramId
+      };
+
+      // Only merge the new profile if none of the users are builders
+      if (selectedProfile === 'new' && mergedUser.builderStatus === null && primaryUser.builderStatus === null) {
```

instead of checking for `mergedUser.builderStatus === null && primaryUser.builderStatus === null` here, i think this method should simply throw an error if it was given invalid input for selectedProfile

---

updated. selected profile of null must be provided if any of the accounts is a builder otherwise we will throw an error